invokeai/app/api/routers/model_records.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2023 Lincoln D. Stein
+"""FastAPI route for model configuration records."""
+
+
+from hashlib import sha1
+from random import randbytes
+from typing import List, Optional
+
+from fastapi import Body, Path, Query, Response
+from fastapi.routing import APIRouter
+from pydantic import BaseModel, ConfigDict, TypeAdapter
+from starlette.exceptions import HTTPException
+
+from invokeai.app.services.model_records import DuplicateModelException, InvalidModelException, UnknownModelException
+from invokeai.backend.model_manager.config import AnyModelConfig, BaseModelType, ModelType
+
+from ..dependencies import ApiDependencies
+
+model_records_router = APIRouter(prefix="/v1/model_records", tags=["model_records"])
+
+ModelConfigValidator = TypeAdapter(AnyModelConfig)
+
+
+class ModelsList(BaseModel):
+    """Return list of configs."""
+
+    models: list[AnyModelConfig]
+
+    model_config = ConfigDict(use_enum_values=True)
+
+
+ModelsListValidator = TypeAdapter(ModelsList)
+
+
+@model_records_router.get(
+    "/",
+    operation_id="list_model_configs",
+    responses={200: {"model": ModelsList}},
+)
+async def list_model_records(
+    base_models: Optional[List[BaseModelType]] = Query(default=None, description="Base models to include"),
+    model_type: Optional[ModelType] = Query(default=None, description="The type of model to get"),
```

Comment:
Should the model_type be non-optional? Or are there times that we just query for all models associated with a base_model?

Reply:
Yes, there are times when we don't specify any args at all in order to get a dump of all models.